NGitLab.Mock/ReleaseCollection.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Linq;
+
+namespace NGitLab.Mock
+{
+    public class ReleaseCollection : Collection<ReleaseInfo>
+    {
+        private Project Project => (Project)Parent;
+
+        public ReleaseCollection(GitLabObject parent)
+            : base(parent)
+        {
+        }
+
+        public ReleaseInfo GetByTagName(string tagName)
+        {
+            return this.FirstOrDefault(r => r.TagName.Equals(tagName));
+        }
+
+        public override void Add(ReleaseInfo release)
+        {
+            if (release is null)
+                throw new ArgumentNullException(nameof(release));
+
+            if (release.TagName == default || Project.Repository.GetTags().FirstOrDefault(t => t.FriendlyName.Equals(release.TagName)) == null)
+            {
+                throw new ArgumentException(nameof(release));
```

Comment:
Please provide a message as first argument, something like `$"Tag '{release.TagName}' invalid or not found"`.